tools/generators/move-packages/index.ts

```diff
@@ -178,3 +179,18 @@ function updateApiExtractor(tree: Tree, schema: MovePackagesGeneratorSchema) {
 
   writeJson(tree, apiExtractorLocalPath, apiExtractorLocal);
 }
+
+function updateBuildLocalScript(tree: Tree, schema: MovePackagesGeneratorSchema) {
+  const { name, destination } = schema;
```

Comment:
this implementation can be simplified to following:

```ts
function updateBuildLocalScript(tree: Tree, schema: MovePackagesGeneratorSchema) {
  const project = getProjectConfig(tree, { packageName: schema.name! });

  updateJson<PackageJson>(tree, project.paths.packageJson, json => {
    // 💡 this is a defensive programming, enforced by TS types 
    json.scripts = json.scripts ?? {};
    const originalScript = json.scripts['build:local'];
    const updatedScript = originalScript.replace(project.normalizedPkgName, schema.destination);
    json.scripts['build:local'] = updatedScript;
    return json;
  });
}
```


**Out of topic:**

I'd suggest to introduce something like 

```ts
interface AssertedSchema extends MovePackagesGeneratorSchema {
  name: string;
}
```

and use that interface for those atomic functions as `name` is required for them to make it work am I right ?

```ts
function updateBuildLocalScript(tree: Tree, schema: AssertedSchema) {
....
```

Reply:
Made the refactor of the `updateBuildLocalScript` function and added an `AssertedSchema` like you suggested in this commit [2d9cd3c](https://github.com/microsoft/fluentui/pull/22667/commits/2d9cd3c0a057df4f763c3842565c93679c2bbcde). I refactored some of the other functions to use the `AssertedSchema` so sorry if the diffs are slightly messy 😅